src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -20,29 +21,35 @@ sealed class BlobFeedAction
         private MSBuild.TaskLoggingHelper Log;
         private static readonly CancellationTokenSource TokenSource = new CancellationTokenSource();
         private static readonly CancellationToken CancellationToken = TokenSource.Token;
+        private const string feedRegex = @"(?<feedurl>https:\/\/(?<accountname>[^\.-]+)(?<domain>[^\/]*)\/((?<token>[a-zA-Z0-9+\/]*?\/\d{4}-\d{2}-\d{2})\/)?(?<containername>[^\/]+)\/(?<relativepath>.*\/)?)index\.json";
+        private string feedUrl;
+        private SleetSource source;
 
         public BlobFeed feed;
-        public int MaxClients { get; set; } = 8;
-        
-        const string feedRegex = @"(?<feedurl>https:\/\/(?<accountname>[^\.-]+)(?<domain>[^\/]*)\/((?<token>[a-zA-Z0-9+\/]*?\/\d{4}-\d{2}-\d{2})\/)?(?<containername>[^\/]+)\/(?<relativepath>.*)\/)index\.json";
 
-        public BlobFeedAction(string expectedFeedUrl, string accountKey, string indexDirectory, MSBuild.TaskLoggingHelper Log)
+        public BlobFeedAction(string expectedFeedUrl, string accountKey, ITaskItem[] itemstoPush, MSBuild.TaskLoggingHelper Log)
         {
             this.Log = Log;
             Match m = Regex.Match(expectedFeedUrl, feedRegex);
-
             if (m.Success)
             {
                 string accountName = m.Groups["accountname"].Value;
                 string containerName = m.Groups["containername"].Value;
                 string relativePath = m.Groups["relativepath"].Value;
-                string feedUrl = m.Groups["feedurl"].Value;
-
-                bool isPublic = string.IsNullOrWhiteSpace(m.Groups["token"].Value);
-                this.feed = new BlobFeed(accountName, accountKey, containerName, relativePath, feedUrl, string.IsNullOrWhiteSpace(indexDirectory) ? Path.GetTempPath() : indexDirectory, Log, isPublic);
+                feed = new BlobFeed(accountName, accountKey, containerName, relativePath, Log);
+                feedUrl = expectedFeedUrl.Replace("index.json", string.Empty);
```

Comment:
I still think the feedUrl should come out from the regex. The goal was we might end up needing the pass the regex as an extra parameter in the build. It is also better then a blind Replace which doesn't take into consideration if "index.json" exists anywhere else in the URL. It is unlikely but still I think good practice to not assume.

Reply:
Oh! Yeah your are right, I did not understand what you meant at the beginning but now I see what you mean